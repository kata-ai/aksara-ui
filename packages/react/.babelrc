{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": ["last 2 chrome versions", "last 2 firefox versions"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "babel-plugin-styled-components"
  ],
  "env": {
    "test": {
      "plugins": ["@babel/plugin-transform-modules-commonjs"]
    }
  }
}
