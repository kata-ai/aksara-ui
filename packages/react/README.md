# @aksara-ui/react

> React implementation of Aksara Design System, from Kata.ai.

This package is part of the Aksara monorepo. Please read the README in [the Aksara repository](https://github.com/kata-ai/aksara-ui) for detailed documentation about this package.
