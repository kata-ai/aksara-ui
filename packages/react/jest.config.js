module.exports = {
  collectCoverageFrom: [
    'src/**/*.{ts,tsx}',
    '!src/**/*.stories.{ts,tsx}',
    '!src/**/*.story.{ts,tsx}',
    '!src/**/index.{ts,tsx}',
    '!src/stories/**/*.{ts,tsx}',
    '!src/utils/storybook/**/*.{ts,tsx}',
    '!src/index.{ts,tsx}',
  ],
  coverageReporters: ['json', 'lcov'],
  moduleDirectories: ['node_modules', '<rootDir>/src'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json'],
  modulePathIgnorePatterns: ['lib', 'lib-es6'],
  setupFilesAfterEnv: ['<rootDir>/test/setupTests.ts'],
  testPathIgnorePatterns: ['<rootDir>/dist/'],
  transform: {
    '^.+\\.(js|jsx|ts|tsx|mjs)$': 'babel-jest',
    '^.+\\.css$': '<rootDir>/config/jest/cssTransform.js',
    '^(?!.*\\.(js|jsx|ts|tsx|mjs|css|json)$)': '<rootDir>/config/jest/fileTransform.js',
  },
};
