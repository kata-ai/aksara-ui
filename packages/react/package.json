{
  "name": "@aksara-ui/react",
  "version": "1.0.1-canary.4",
  "description": "React implementation of Aksara Design System, from Kata.ai.",
  "main": "dist/index.js",
  "module": "dist/core.esm.js",
  "typings": "dist/index.d.ts",
  "aksara:src": "./src/index.ts",
  "repository": "https://github.com/kata-ai/wicara.git",
  "author": "Kata.ai <dev@kata.ai>",
  "license": "Apache-2.0",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsdx build --transpileOnly --tsconfig tsconfig.build.json",
    "build-storybook": "build-storybook --quiet",
    "clean": "rimraf dist",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
    "prepublish": "yarn build",
    "report-coverage": "codecov -f ./coverage/*.json",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose",
    "storybook": "start-storybook -p 9001",
    "storybook:build": "build-storybook --quiet",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "type-check": "tsc --emitDeclarationOnly"
  },
  "dependencies": {
    "@aksara-ui/core": "^1.0.1-canary.1",
    "@popperjs/core": "^2.10.2",
    "@reach/accordion": "^0.16.1",
    "@reach/visually-hidden": "^0.16.0",
    "@styled-system/css": "^5.1.5",
    "@styled-system/should-forward-prop": "^5.1.5",
    "@styled-system/theme-get": "^5.1.2",
    "@types/styled-system": "^5.1.13",
    "@types/styled-system__css": "^5.0.16",
    "@types/styled-system__theme-get": "^5.0.1",
    "clsx": "^1.1.1",
    "csstype": "^3.0.9",
    "deepmerge": "^4.2.2",
    "focus-trap": "^6.6.0",
    "polished": "^4.1.3",
    "react-fast-compare": "^3.2.0",
    "react-popper": "^2.2.5",
    "react-popper-tooltip": "^4.3.0",
    "react-transition-group": "^4.4.2",
    "styled-system": "^5.1.4",
    "tslib": "^2.3.1"
  },
  "peerDependencies": {
    "@aksara-ui/icons": "^1.0.0-canary.0",
    "react": "^16.10.0 || ^17.0.0",
    "react-dom": "^16.10.0 || ^17.0.0",
    "styled-components": "^5.1.0"
  },
  "devDependencies": {
    "@aksara-ui/icons": "^1.0.0",
    "@babel/core": "^7.15.0",
    "@storybook/addon-actions": "^6.3.7",
    "@storybook/addon-essentials": "^6.3.7",
    "@storybook/addon-knobs": "^6.3.0",
    "@storybook/addon-links": "^6.3.7",
    "@storybook/react": "^6.3.7",
    "@storybook/theming": "^6.3.7",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@types/jest": "^26.0.24",
    "@types/react-transition-group": "^4.4.2",
    "@types/styled-components": "^5.1.12",
    "babel-loader": "^8.2.2",
    "babel-plugin-styled-components": "^1.13.2",
    "babel-preset-react-app": "^10.0.0",
    "codecov": "^3.8.3",
    "copee": "^1.0.6",
    "jest": "^26.6.3",
    "jest-styled-components": "^7.0.5",
    "react": "^17.0.2",
    "react-docgen-typescript-loader": "^3.6.0",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.0"
  }
}
