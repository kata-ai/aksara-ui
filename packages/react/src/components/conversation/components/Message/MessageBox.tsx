// TODO
// container for message
// handle like UI
// type will be used as extended type for other type of message
// so other message will be derived the props just like `like status`
