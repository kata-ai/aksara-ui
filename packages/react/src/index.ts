export * from './aksara-provider';
export * from './components';
export * from './layout';
export * from './theme';
export * from './typography';
export * from '@aksara-ui/helpers';
