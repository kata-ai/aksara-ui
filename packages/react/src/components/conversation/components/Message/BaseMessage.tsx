import * as React from 'react';
import { Box } from '../../../../layout';
import { Text } from '../../../../typography';
import { useComponentStyles } from '../../../../system';

export type BaseMessageVariant = 'inbound' | 'outbound';

export interface BaseMessageProps {
  variant: BaseMessageVariant;
  time?: Date;
  read?: boolean;
  like?: boolean;
}

export const BaseMessage: React.FC<BaseMessageProps> = ({ children, variant, time, read }) => {
  const BaseStyle = useComponentStyles('Base', { variant });
  const timeLabelStyle = useComponentStyles('timeLabel', { variant });

  return (
    <Box sx={BaseStyle}>
      <Box>{children}</Box>
      {time && (
        <Text
          fontSize={10}
          lineHeight={16}
          fontWeight={400}
          sx={timeLabelStyle}
        >{`${time.getHours}:${time.getMinutes}`}</Text>
      )}
      {read && 'tes'}
    </Box>
  );
};
