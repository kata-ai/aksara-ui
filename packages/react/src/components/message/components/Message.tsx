import { IconClose, IconInfo, IconTickRounded, IconWarningRounded, IconWarningTriangle } from '@aksara-ui/icons';
import * as React from 'react';
import { Box } from '../../../layout';
import { useComponentStyles } from '../../../system';
import { Heading, Paragraph } from '../../../typography';
import { UnstyledButton } from '../../button';

export type MessageStates = 'default' | 'info' | 'warning' | 'critical' | 'success';

export interface MessageProps {
  /** Message box content. Could be a string or a `ReactNode`. */
  message: string | React.ReactNode;
  /** Message title. */
  title?: string;
  /** Additional CSS classes to give to the banner. */
  className?: string;
  /** Additional CSS properties to give to the banner. */
  style?: React.CSSProperties;
  /** The message variant. Defaults to `default`. */
  variant?: MessageStates;
  /** Triggers when the Close button is clicked. */
  onClose?: () => void;
}

/** Banners are used to convey important information to users. */
const Message: React.FC<MessageProps> = ({
  className,
  style,
  title,
  message,
  variant = 'default',
  onClose,
  ...rest
}) => {
  const messageBaseStyles = useComponentStyles('messageBase', { variant });
  const messageIconStyles = useComponentStyles('messageIcon', { variant });

  const handleClose = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();

    if (onClose) {
      onClose();
    }
  };

  const renderIconStates = () => {
    switch (variant) {
      case 'default': {
        return <IconInfo size={24} aria-hidden fill="currentColor" />;
      }
      case 'success': {
        return <IconTickRounded size={24} aria-hidden fill="currentColor" />;
      }
      case 'info': {
        return <IconInfo size={24} aria-hidden fill="currentColor" />;
      }
      case 'warning': {
        return <IconWarningTriangle size={24} aria-hidden fill="currentColor" />;
      }
      case 'critical': {
        return <IconWarningRounded size={24} aria-hidden fill="currentColor" />;
      }
      default: {
        return null;
      }
    }
  };

  const renderMessage = () => {
    if (typeof message !== 'string') {
      return message;
    }

    return (
      <Paragraph display="inline-block" scale={200} m={0}>
        {message}
      </Paragraph>
    );
  };

  return (
    <Box className={className} style={style} sx={messageBaseStyles} {...rest}>
      <Box display="flex" alignItems="flex-start" justifyContent="center" py="md" pl="md" sx={messageIconStyles}>
        {renderIconStates()}
      </Box>
      <Box flex="1 1 auto" p="md">
        {title ? (
          <Heading as="h4" mb="xs" scale={300} fontWeight={600}>
            {title}
          </Heading>
        ) : null}
        {renderMessage()}
      </Box>
      {onClose && (
        <Box display="flex" alignItems="flex-start" justifyContent="center" p="md" color="greymed04">
          <UnstyledButton
            display="flex"
            alignItems="center"
            justifyContent="center"
            width={24}
            height={24}
            type="button"
            aria-label="Close"
            data-testid="Banner-closeButton"
            onClick={handleClose}
          >
            <IconClose size={24} aria-hidden fill="currentColor" />
          </UnstyledButton>
        </Box>
      )}
    </Box>
  );
};

Message.displayName = 'Message';

export default Message;
