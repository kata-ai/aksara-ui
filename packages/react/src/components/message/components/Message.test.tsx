import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Message from './Message';

const handleClose = jest.fn();

describe('components/Banner', () => {
  describe('<Banner />', () => {
    test('renders correctly', () => {
      const { container } = render(<Message state="success" message="test banner" />);

      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders with title', () => {
      const { getByText } = render(<Message state="success" title="test title" message="test banner" />);

      const titleText = getByText('test title');
      expect(titleText).toBeInTheDocument();
    });

    test('renders close button with onClose', () => {
      const { getByTestId } = render(<Message state="success" message="test banner" onClose={handleClose} />);

      const closeButton = getByTestId('Banner-closeButton');
      expect(closeButton).toBeInTheDocument();
    });

    test('fires the onClose event', () => {
      const { getByTestId } = render(<Message state="success" message="test banner" onClose={handleClose} />);

      const closeButton = getByTestId('Banner-closeButton');
      fireEvent.click(closeButton);
      expect(handleClose).toBeCalledTimes(1);
    });
  });
});
