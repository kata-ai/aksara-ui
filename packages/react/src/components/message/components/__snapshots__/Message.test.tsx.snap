// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Banner <Banner /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: md;
  padding-bottom: md;
  padding-left: md;
}

.c1 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  padding: md;
}

.c2 {
  display: inline-block;
  margin: 0;
  -webkit-letter-spacing: -0.05px;
  -moz-letter-spacing: -0.05px;
  -ms-letter-spacing: -0.05px;
  letter-spacing: -0.05px;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <svg
      aria-hidden="true"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
    >
      <path
        d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10 10-4.5 10-10S17.5 2 12 2zm4.2 8.2l-5 5c-.4.4-1 .4-1.4 0l-2-2c-.4-.4-.4-1 0-1.4.4-.4 1-.4 1.4 0l1.3 1.3 4.3-4.3c.4-.4 1-.4 1.4 0 .4.4.4 1 0 1.4z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      test banner
    </p>
  </div>
</div>
`;
